src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessWaitState.Unix.cs

```diff
@@ -324,7 +322,7 @@ internal bool GetExited(out int? exitCode, bool refresh)
 
                 // Is another wait operation in progress?  If so, then we haven't exited,
                 // and that task owns the right to call CheckForNonChildExit.
-                if (_waitInProgress != null)
+                if (_waitInProgress is Task waitInProgress && !waitInProgress.IsCompleted)
```

Comment:
isn't `_waitInProgress is Task waitInProgress` only meant to null-check?
```suggestion
                if (_waitInProgress != null && !_waitInProgress.IsCompleted)
```

Reply:
Alternatively, it could be initialized with a `CompletedTask` and the `null` no longer needed to be handled.